program/metaplex/tokenmeta/instruction.go

```diff
@@ -0,0 +1,72 @@
+package tokenmeta
+
+import (
+	"github.com/near/borsh-go"
+	"github.com/pkg/errors"
+	"github.com/portto/solana-go-sdk/common"
+	"github.com/portto/solana-go-sdk/types"
+)
+
+type Instruction uint8
+
+const (
+	InstructionCreateMetadataAccount Instruction = iota
+)
+
+func CreateMetadataAccount(metadata, mint, mintAuthority, payer, updateAuthority common.PublicKey, updateAuthorityIsSigner, isMutable bool, mintData Data) (types.Instruction, error) {
+	data, err := borsh.Serialize(struct {
+		Instruction Instruction
+		Data        Data
+		IsMutable   bool
+	}{
+		Instruction: InstructionCreateMetadataAccount,
+		Data:        mintData,
+		IsMutable:   isMutable,
+	})
+
+	if err != nil {
+		return types.Instruction{}, errors.Wrap(err, "failed serialize")
+	}
+
+	return types.Instruction{
+		ProgramID: common.MetaplexTokenMetaProgramID,
+		Accounts: []types.AccountMeta{
+			{
+				PubKey:     metadata,
+				IsSigner:   false,
+				IsWritable: true,
+			},
+			{
+				PubKey:     mint,
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			{
+				PubKey:     mintAuthority,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+			{
+				PubKey:     payer,
+				IsSigner:   true,
+				IsWritable: false,
```

Comment:
according to https://github.com/metaplex-foundation/metaplex/blob/master/rust/token-metadata/program/src/instruction.rs#L270
I think it should be
```diff
PubKey:     payer,
IsSigner:   true,
- IsWritable: false,
+ IsWritable: true,
```

Reply:
Very strange thing - it have worked for me on testnet. I've fixed it, it still works